dm/pkg/ha/bound.go

```diff
@@ -269,7 +269,11 @@ func WatchSourceBound(ctx context.Context, cli *clientv3.Client, worker string,
 		case <-ctx.Done():
 			return
 		case resp, ok := <-ch:
```

Comment:
what's the reason of this WatchChan being closed?

Reply:
when networkw is poor, all grpc stream is dead, etcd watch will retrn an closed channel

see here https://github.com/etcd-io/etcd/blob/20c89df5e5e2d738efb9c276d954d754eb86918b/client/v3/watch.go#L325

and this [issue](https://github.com/etcd-io/etcd/issues/8980#issuecomment-350070282)

Reply:
The link you given is not the version and path of etcd we use. 

https://github.com/pingcap/tiflow/blob/b3014061250cb3598c491624135217d4cd76d386/go.mod#L83

Reply:
w.streams will only be nil after `w.Close()`, I don't think that's the root cause

Reply:
if you have time can you help locate the problem, I don't have any ideas here at the moment, I'll send you the log in private chat~